src/tactic/core.lean

```diff
@@ -2306,3 +2306,39 @@ Otherwise, it fails.
 meta def list.find_defeq (red : tactic.transparency) {v} (m : list (expr × v)) (e : expr) :
   tactic (expr × v) :=
 m.mfind $ λ ⟨e', val⟩, tactic.is_def_eq e e' red
+
+/-!
```

Comment:
I think `tactic.core` is an odd place for this. A lot of the notations are for tactics, but not all. Would it be better in a dedicated file as high in the import hierarchy as we can get it? (Imported by `tactic.core` or `logic.basic` or whatever.)

Reply:
Yeah, I think it does make more sense to have a dedicated file. How about `tactic.notations`, imported by `tactic.core`?

Reply:
I settled on `tactic.reserved_notation`, and I added it to the imports of `logic.basic` and `logic.relator`. 

I'll push the commit after today's nolints update.